src/lib/server/postgres_wire_handler.cpp

```diff
@@ -0,0 +1,154 @@
+  auto length = ntohl(n_length);
+
+  auto n_version = read_value<uint32_t>(packet);
+  auto version = ntohl(n_version);
+
+  // Reset data buffer
+  packet.offset = packet.data.cbegin();
+
+  // Special SSL version number
+  if (version == 80877103) {
+    return 0;
+  } else {
+    // Subtract read bytes from total length
+    return length - (2 * sizeof(uint32_t));
+  }
+}
+
+void PostgresWireHandler::handle_startup_package_content(const InputPacket& packet) {
+  // Ignore the content for now
+  read_values<char>(packet, packet.data.size());
+}
+
+RequestHeader PostgresWireHandler::handle_header(const InputPacket& packet) {
+  auto tag = read_value<NetworkMessageType>(packet);
+
+  auto n_length = read_value<uint32_t>(packet);
+  auto length = ntohl(n_length);
+
+  packet.offset = packet.data.begin();
+
+  // Return length minus the already read bytes (the message type doesn't count into the length)
+  return {/* message_type = */ tag, /* payload_length = */ static_cast<uint32_t>(length - sizeof(n_length))};
+}
+
+std::string PostgresWireHandler::handle_query_packet(const InputPacket& packet) { return read_string(packet); }
+
+ParsePacket PostgresWireHandler::handle_parse_packet(const InputPacket& packet) {
+  auto statement_name = read_string(packet);
+
+  auto query = read_string(packet);
+
+  auto n_parameter_data_types = read_value<uint16_t>(packet);
+
+  /*auto parameter_data_types = */ read_values<uint32_t>(packet, n_parameter_data_types);
+
+  //  SQLPipeline sql_pipeline{query};
+  //  auto parsed_statements = sql_pipeline.get_parsed_sql_statements();
+  //  if (parsed_statements.size() != 1) {
+  //    throw std::runtime_error("Only exactly 1 statement supported.");
+  //  }
+
+  return ParsePacket{std::move(statement_name), std::move(query)};  //, std::move(sql_pipeline)};
+}
+
+BindPacket PostgresWireHandler::handle_bind_packet(const InputPacket& packet) {
+  auto portal = read_string(packet);
+
+  auto statement_name = read_string(packet);
+
+  auto n_format_codes = read_value<int16_t>(packet);
+
+  auto format_codes = read_values<int16_t>(packet, n_format_codes);
+
+  auto n_parameter_values = ntohs(read_value<int16_t>(packet));
+
+  std::vector<AllParameterVariant> parameter_values;
+  for (auto i = 0; i < n_parameter_values; ++i) {
+    auto parameter_value_length = ntohl(read_value<int32_t>(packet));
+    auto x = read_values<char>(packet, parameter_value_length);
+    const std::string x_str(x.begin(), x.end());
+    parameter_values.emplace_back(x_str);
+  }
+
+  auto n_result_column_format_codes = read_value<int16_t>(packet);
+  auto result_column_format_codes = read_values<int16_t>(packet, n_result_column_format_codes);
+
+  return BindPacket{statement_name, portal, std::move(parameter_values)};
+}
+
+std::string PostgresWireHandler::handle_execute_packet(const InputPacket& packet) {
+  const auto portal = read_string(packet);
+  /*const auto max_rows = */ read_value<int32_t>(packet);
```

Comment:
we chose `read_value` because this naming is also used in the csv/binary import. then we have a consistent naming for the type of behaviour within hyrise.